src/main/java/org/apache/commons/scxml2/io/SCXMLReader.java

```diff
@@ -1094,7 +1094,21 @@ private static void readData(final XMLStreamReader reader, final Configuration c
 
         Data datum = new Data();
         datum.setId(readRequiredAV(reader, ELEM_DATA, ATTR_ID));
-        datum.setExpr(readAV(reader, ATTR_EXPR));
+        final String expr = readAV(reader, ATTR_EXPR);
+        final String src = readAV(reader, ATTR_SRC);
+
+        if (expr != null && src != null) {
+          LogFactory.getLog(SCXMLReader.class).error(
+              "Found src and expr attributes for data node '" + datum.getId() + "', which is not valid SCXML.");
+          throw new ModelException();
```

Comment:
I think you should invoke #reportConflictingAttribute(reader, configuration, ...) instead here. That method leaves logs in a consistent manner and can handle strict or non-strict mode properly.